scanner/decisions/0010-separate-versioning-for-vuln-bundles.md

```diff
@@ -0,0 +1,175 @@
+
+The Scanner Matcher, using ClairCore, runs the Vulnerability Updater (aka. Updater), which fetches the bundle, parses the data, and updates the vulnerability database.
+
+Suppose a backward-incompatible schema change is made in ClairCore and incorporated into Scanner.  In that case, the following Exporter execution will update the bundle adhering to the new schema.  Existing Scanner instances can only parse the changed data if they upgrade to the new ClairCore version that understands the schema changes.
+
+Due to the tight coupling between the ClairCore version and schema, one bundle per StackRox release was introduced for Scanner V4.
+
+Each bundle is created with an Exporter running the same ClairCore version used to build the release, ensuring compatibility.  The Exporter serves bundles at `https://definitions.stackrox.io/v4/<version>/vulnerabilities.zip` where `<version>` is the supported StackRox release (such as "4.5.0").  Scanner uses its release version tagged during build time to determine which URL to fetch from.
+
+In offline mode, one offline bundle per release was introduced for Scanner V4, coexisting with the "latest" Scanner V2 offline bundles.  To facilitate user migration to per-release offline bundles, the latest offline bundle used by Scanner V2 (at the time of this writing served at https://install.stackrox.io/scanner/scanner-vuln-updates.zip) was changed to contain Scanner V4's offline bundles for `4.[45][.x]` releases.  A command `roxctl scanner download-db` was created to facilitate users fetching the right offline bundle for their release.
+
+This approach needs to be revised for the following reasons:
+
+1.  The vulnerability schema rarely changes, and maintaining multiple bundles is expensive.  It requires multiple GitHub workflow runs, which increases the chances of failure and pressure on the security sources' endpoints.
+2.  It can be challenging to backport fixes and updates to previous releases.  This requires patch releases, which trigger the release workflow.  Improving the release process can be complex, as it may require patch release cuts, changes to different branches, and integration between logic that is managed in the master branch, executing other releases binaries.
+3.  The latest offline bundles does not support post `4.5` releases.  Bundle embedding for each release increases the cost to store and consume offline bundles.  The `roxctl` interface and documentation today needs to be clearer on these limitations.
+
+## Decision
+
+We will implement a versioning stream for vulnerability bundles, bumped when we determine that ClairCore's schema has changed.
```

Comment:
related to above, but to be clear, I do believe there are non-schema-related changes which may cause this bump

Reply:
I've added a footnote to clarify.

Reply:
A footnote was added to clarify this.